mnefun/_mnefun.py

```diff
@@ -1368,10 +1368,10 @@ def do_preprocessing_combined(p, subjects):
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
-            assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict
```

Comment:
Cool. I wonder if we should also raise an error or issue a warning if they are both `None` but `p.auto_bad is not None`, though, because in that case you're asking for auto bad detection but passing reject and flat params that do nothing. WDYT?